packages/components/src/resizable-box/resize-tooltip/utils.ts

```diff
@@ -9,7 +9,8 @@ import useResizeAware from 'react-resize-aware';
  */
 import { useEffect, useRef, useState } from '@wordpress/element';
 
-const { clearTimeout, setTimeout } = window;
+const { clearTimeout = () => undefined, setTimeout = () => undefined } =
```

Comment:
Once this PR is rebased the `noop` function will no longer be imported from `lodash` but [this file will define its own](https://github.com/WordPress/gutenberg/blob/b3c6f67a1ff8cb9e5f40a4b9c83a4bf866e0707c/packages/components/src/resizable-box/resize-tooltip/utils.ts#L8). If we change the `noop` definition from `() => {}` to `() => undefined` then you'll be able to set the fallback values for `clearTimeout` and `setTimeout` to the shared `noop` function.

Reply:
It was suggested internally to use `globalThis` here instead of window. Props @jsnajdr I think.

e.g. something like:

```suggestion
const { clearTimeout, setTimeout } = globalThis;
```

Reply:
Apparently we can't use the `globalThis` trick in cases where we need to use the return of the `setTimeout` call, because in Node it returns `NodeJS.Timeout` instead of `number`.

https://github.com/WordPress/gutenberg/blob/87689226f9159405653e5358835c0d1d2d094b96/packages/components/src/resizable-box/resize-tooltip/utils.ts#L109

In this code it won't ever actually run in a SSR context because the call is inside a `useEffect`, but the TypeScript checker doesn't know that, so it kindly throws an error 😄

Reply:
Would something [like this](https://stackoverflow.com/a/56239226) work?

```
const moveTimeoutRef = useRef< ReturnType< typeof setTimeout > >()
```

[Potentially](https://stackoverflow.com/a/70960803/) might need some nulls.

Reply:
Can't we just refactor the code so that we can write `window.setTimeout` and `window.clearTimeout` inline inside the `setTimeout` ?